libraries/stdlib/samples/test/samples/collections/builders.kt

```diff
@@ -0,0 +1,77 @@
+package samples.collections
+
+import samples.*
+import kotlin.test.*
+
+@Suppress("ReplaceAssertBooleanWithAssertEquality", "UNRESOLVED_REFERENCE" /* See KT-30129 */) // TODO: remove this in 1.4
+@RunWith(Enclosed::class)
+class Builders {
+    class Lists {
+        @Sample
+        fun buildListSample() {
+            val x = buildList {
+                add('a')
+                add('b')
+            }
+            assertPrints(x, "[a, b]")
+
+            val y = buildList(x.size + 1) {
+                addAll(x)
+                add('c')
+            }
+            assertPrints(y, "[a, b, c]")
+
+            assertFails("initial capacity cannot be negative") {
+                buildList(-1) {
+                    add('a')
+                }
+            }
+        }
+    }
+
+    class Sets {
+        @Sample
+        fun buildSetSample() {
+            val x = buildSet {
+                add('a')
+                add('b')
+            }
+            assertPrints(x.sorted(), "[a, b]")
```

Comment:
`buildSet` and `buildMap` preserve the insertion order, so sorting is not needed here.